File: code/backend/DarkDeeds.TaskServiceApp.Entities/Models/TaskEntity.cs
using System;
using DarkDeeds.TaskServiceApp.Entities.Enums;
using DarkDeeds.TaskServiceApp.Entities.Models.Interfaces;

namespace DarkDeeds.TaskServiceApp.Entities.Models
{
    public class TaskEntity : IUserOwnedEntity
    {
        public string Uid { get; set; }
        public string Title { get; set; }
        public int Order { get; set; }
        public DateTime? Date { get; set; }
        public TaskTypeEnum Type { get; set; }
        public int? Time { get; set; }
        public bool IsCompleted { get; set; }
        public bool IsProbable { get; set; }
        public int Version { get; set; }
        public bool IsDeleted { get; set; }
        public string UserId { get; set; }
    }
}

File: code/frontend4/src/models/entities/task-entity.ts
import { TaskTypeEnum } from '../enums/task-type-enum'

export interface TaskEntity {
    uid: string
    title: string
    date: Date | null
    order: number
    changed: boolean
    completed: boolean
    deleted: boolean
    type: TaskTypeEnum
    isProbable: boolean
    version: number
    time: number | null
}
